cmake/alpakaCommon.cmake

```diff
@@ -0,0 +1,906 @@
+                        message(WARNING "${CMAKE_CXX_COMPILER_ID} ${CMAKE_CXX_COMPILER_VERSION} does not support -G with CUDA! "
+                                        "Device debug symbols NOT added.")
+                    else()
+                        list(APPEND CUDA_NVCC_FLAGS -G)
+                    endif()
+                endif()
+
+                if(ALPAKA_CUDA_FAST_MATH)
+                    list(APPEND CUDA_NVCC_FLAGS --use_fast_math)
+                endif()
+
+                if(ALPAKA_CUDA_FTZ)
+                    list(APPEND CUDA_NVCC_FLAGS --ftz=true)
+                else()
+                    list(APPEND CUDA_NVCC_FLAGS --ftz=false)
+                endif()
+
+                if(ALPAKA_CUDA_SHOW_REGISTER)
+                    list(APPEND CUDA_NVCC_FLAGS -Xptxas=-v)
+                endif()
+
+                # Always add warning/error numbers which can be used for suppressions
+                list(APPEND CUDA_NVCC_FLAGS -Xcudafe --display_error_number)
+
+                # avoids warnings on host-device signatured, default constructors/destructors
+                list(APPEND CUDA_NVCC_FLAGS -Xcudafe --diag_suppress=esa_on_defaulted_function_ignored)
+
+                # avoids warnings on host-device signature of 'std::__shared_count<>'
+                if(CUDA_VERSION EQUAL 10.0)
+                    list(APPEND CUDA_NVCC_FLAGS -Xcudafe --diag_suppress=2905)
+                elseif(CUDA_VERSION EQUAL 10.1)
+                    list(APPEND CUDA_NVCC_FLAGS -Xcudafe --diag_suppress=2912)
+                elseif(CUDA_VERSION EQUAL 10.2)
+                    list(APPEND CUDA_NVCC_FLAGS -Xcudafe --diag_suppress=2976)
+                endif()
+
+                if(ALPAKA_CUDA_KEEP_FILES)
+                    file(MAKE_DIRECTORY "${PROJECT_BINARY_DIR}/nvcc_tmp")
+                    list(APPEND CUDA_NVCC_FLAGS --keep --keep-dir ${PROJECT_BINARY_DIR}/nvcc_tmp)
+                endif()
+
+                option(ALPAKA_CUDA_SHOW_CODELINES "Show kernel lines in cuda-gdb and cuda-memcheck" OFF)
+                if(ALPAKA_CUDA_SHOW_CODELINES)
+                    list(APPEND CUDA_NVCC_FLAGS --source-in-ptx -lineinfo)
+                    if(NOT MSVC)
+                        list(APPEND CUDA_NVCC_FLAGS -Xcompiler -rdynamic)
+                    endif()
+                    set(ALPAKA_CUDA_KEEP_FILES ON CACHE BOOL "activate keep files" FORCE)
+                endif()
+            endif()
+
+            if(OpenMP_CXX_FOUND)
+                # correctly propagate OpenMP flags
+                # This can be removed once we support CMake's first class CUDA support.
+                target_compile_options(alpaka INTERFACE ${OpenMP_CXX_FLAGS})
+            endif()
+
+            target_link_libraries(alpaka INTERFACE ${CUDA_CUDART_LIBRARY})
+            target_include_directories(alpaka INTERFACE ${CUDA_INCLUDE_DIRS})
+
+            # export NVCC flags to parent scope in case alpaka is another project's subdirectory
```

Comment:
must be 
```
# export to parent scope if used a sub_directory
if(NOT ${CMAKE_PROJECT_NAME} STREQUAL ${PROJECT_NAME})
    # export HIPCC flags to parent scope in case alpaka is another project's subdirectory
    set(HIP_HIPCC_FLAGS ${HIP_HIPCC_FLAGS} PARENT_SCOPE)
    set(HIP_NVCC_FLAGS ${HIP_NVCC_FLAGS} PARENT_SCOPE)
    set(HIP_VERBOSE_BUILD ${HIP_VERBOSE_BUILD} PARENT_SCOPE)
endif()
```

to avoid the warning if no parent scope exists